lib/membrane_mp4/demuxer/isom.ex

```diff
@@ -332,12 +347,13 @@ defmodule Membrane.MP4.Demuxer.ISOM do
 
     # Parse the data we received so far (partial or the whole mdat box in a single buffer) and
     # either store or send the data (if all pads are connected)
+    mdat_header_size = state.mdat_header_size
 
     data =
       if Keyword.has_key?(state.boxes, :mdat) do
         state.boxes[:mdat].content
       else
-        <<_header::binary-size(@header_size), content::binary>> = state.partial
+        <<_header::binary-size(mdat_header_size), content::binary>> = state.partial
```

Comment:
maybe it would be better to parse the header with `Header.parse` instead of keeping the header size in state

Reply:
My idea was to keep the size of the whole mdat (header + content) instead of header and content size separately.